uaclient/update_info.py

```diff
@@ -0,0 +1,90 @@
+from typing import Any, Dict
+
+from uaclient import util
+from uaclient.config import UAConfig
+
+# We are doing this because we are sure that python3-apt comes with the distro,
+# but it cannot be installed in a virtual environment to be properly tested.
+# All our unit tests are mocking the Cache object, so we are not losing
+# anything here. If by any chance there is no python3-apt in the future, the
+# integration tests should be able to detect it through failure, because we
+# don't expect an empty apt cache there.
+try:
+    from apt import Cache  # type: ignore
+except ImportError:
+
+    def Cache():
+        return []
+
+
+ESM_SERVICES = ("esm-infra", "esm-apps")
+
+series = util.get_platform_info()["series"]
+
+
+def archive_to_service(archive):
+    if archive in (
+        "{}-infra-updates".format(series),
+        "{}-infra-security".format(series),
+    ):
+        return "esm-infra"
+    if archive in (
+        "{}-apps-updates".format(series),
+        "{}-apps-security".format(series),
+    ):
+        return "esm-apps"
+    return archive
```

Comment:
This kind of function is perfect for an easy parametrized unit test, though the `series` variable at the module level might complicate things. You could call that down in `update_info` and pass it as a param instead